### PR TITLE
Implement app for programming and debugging NFC chips

### PR DESCRIPTION
**Pull request by TechnoElf**

### CODE REVIEW

File: app/chip_debug/src/main/res/mipmap-hdpi/ic_launcher.webp
Comment (TheJJ) [inline app/chip_debug/src/main/res/mipmap-hdpi/ic_launcher.webp:1]: can we leave out these placeholder images? :)

File: app/chip_debug/src/test/java/de/stustapay/chip_debug/ExampleUnitTest.kt
Comment (TheJJ) [inline app/chip_debug/src/test/java/de/stustapay/chip_debug/ExampleUnitTest.kt:1]: obsolete file

File: app/chip_debug/src/main/res/drawable/ic_launcher_background.xml
Comment (TheJJ) [inline app/chip_debug/src/main/res/drawable/ic_launcher_background.xml:1]: obsolete file

File: app/chip_debug/src/androidTest/java/de/stustapay/chip_debug/ExampleInstrumentedTest.kt
Comment (TheJJ) [inline app/chip_debug/src/androidTest/java/de/stustapay/chip_debug/ExampleInstrumentedTest.kt:1]: obsolete file

File: app/chip_debug/build.gradle [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    debugImplementation "androidx.compose.ui:ui-test-manifest:$compose_ui_version"
+    implementation "com.google.dagger:hilt-android:$hilt_version"
+    kapt "com.google.dagger:hilt-compiler:$hilt_version"
+    kapt "androidx.hilt:hilt-compiler:$hilt_android_version"
+    annotationProcessor "androidx.hilt:hilt-compiler:$hilt_android_version"
+    implementation 'androidx.hilt:hilt-navigation-compose:1.0.0'
+    implementation "androidx.lifecycle:lifecycle-viewmodel-compose:$lifecycle_version"
+    implementation "androidx.lifecycle:lifecycle-runtime-compose:$lifecycle_version"
+    implementation "com.ionspin.kotlin:bignum:0.3.9"
+    implementation "com.ionspin.kotlin:bignum-serialization-kotlinx:0.3.2"
```
Comment (TheJJ) [inline app/chip_debug/build.gradle:117]: many of the dependencies are not needed for the chipdebugapp

File: app/app/src/main/java/de/stustapay/stustapay/ui/account/AccountProperties.kt [hunk 2]
```diff
@@ -125,7 +125,7 @@ fun AccountProperties(
                 Divider()
                 SummaryEntry(
                     stringResource(R.string.customer_restriction),
-                    when (account.restriction.value) {
+                    when (account.restriction!!.value) {
```
Comment (TheJJ) [inline app/app/src/main/java/de/stustapay/stustapay/ui/account/AccountProperties.kt:128]: these forced evaluations are timebombs for app-crashes - why do we need them now?

**Review by TheJJ — COMMENTED** [review]
the split in a separate debuggingapp is good, as long as we keep all the implemented nfc features in the separate library :)
i'd leave out some of the example files.
the chip manufacturer will surely be happy about a dedicated app for them :)

**Comment by TechnoElf** [comment]
> the split in a separate debuggingapp is good, as long as we keep all the implemented nfc features in the separate library :) i'd leave out some of the example files. the chip manufacturer will surely be happy about a dedicated app for them :)

This is still wip - my intention would be to extract all common code (so probably most non-ui code) into a library that can be shared between all apps

**Review by TheJJ — APPROVED** [review]

**Pull request merged**